website/docs/components/table/partials/code/how-to-use.md

```diff
@@ -545,6 +545,153 @@ In this case the table layout is still set to `auto` (default). If instead you w
+```handlebars
+<div class="doc-table-multiselect-with-pagination-demo">
+  <Hds::Table
+    @isSelectable={{true}}
+    @onSelectionChange={{this.demoOnSelectionChangeWithPagination}}
+    @model={{this.demoPaginatedData}}
+    @columns={{array
+      (hash key="artist" label="Artist")
+      (hash key="album" label="Album")
+      (hash key="year" label="Year")
+    }}
+  >
+    <:body as |B|>
+      <B.Tr @selectionKey={{B.data.id}} @isSelected={{B.data.isSelected}} @selectionAriaLabelSuffix="row {{B.data.artist}} / {{B.data.album}}">
+        <B.Td>{{B.data.artist}}</B.Td>
+        <B.Td>{{B.data.album}}</B.Td>
+        <B.Td>{{B.data.year}}</B.Td>
+      </B.Tr>
+    </:body>
+  </Hds::Table>
+  <Hds::Pagination::Numbered
+    @totalItems={{this.demoTotalItems}}
+    @currentPage={{this.demoCurrentPage}}
+    @pageSizes={{array 2 4}}
+    @currentPageSize={{this.demoCurrentPageSize}}
+    @onPageChange={{this.demoOnPageChange}}
+    @onPageSizeChange={{this.demoOnPageSizeChange}}
+    @ariaLabel="Pagination for multi-select table"
+  />
+</div>
+```
+
+Depending on the expected behavior, you will need to implement the consumer-side logic that handles the persistence (or not) using the `@onSelectionChange` callback function. For the example above, something like this:
+
+```javascript
+@action
+demoOnSelectionChangeWithPagination({ selectableRowsStates }) {
+  // we loop over all the displayed table rows (a subset of the dataset)
+  selectableRowsStates.forEach((row) => {
+    // we find the record in the dataset corresponding to the current row
+    const recordToUpdate = this.demoSourceData.find(
+      (modelRow) => modelRow.id === row.selectionKey
+    );
+    if (recordToUpdate) {
+      // we update the record `isSelected` state based on the row (checkbox) state
+      recordToUpdate.isSelected = row.isSelected;
+    }
+  });
+}
+
+```
+
+For details about the arguments provided to the `@onSelectionChange` callback function refer to the [Component API](#component-api) section.
+
+
+#### Usability and accessibility considerations
+
+Since the "selected" state of a row is communicated visually via the checkbox selection and for screen-reader users via the `aria-label` applied to the checkbox, there are some important considerations to keep in mind when implementing a multi-select table.
+
+If the selection status of the rows is persisted even when a row is not displayed in the UI, you need to be mindful of what the expectations of the user may be: how do we make them aware that the action that they are going to perform may involve table rows that were previously selected by the user but that are not displayed in that moment?
+
+Even more complex is the case of the "Select all" checkbox. While it's pretty straightforward the expected behaviour when all the table rows are displayed on the table (precisely, "select all the rows in the table"), it may not be obvious what is the expected behaviour when the table rows are paginated of filtered.
```

Comment:
I think this needs a bit of cleanup, maybe consider something like:

> Even more complex is the case of the "Select all" checkbox in the table header. While the expected behavior might seem straightforward when all rows are displayed in the table, it may not be obvious what the expected behavior is when the table rows are paginated or have been filtered.